rdtools/test/analysis_test.py

```diff
@@ -0,0 +1,246 @@
+    normalized_daily = normalized_daily + noise
+
+    return normalized_daily
+
+
+def test_srr_soiling():  # duplicate of test_srr_soiling_fixture
+    reps = 10
+    
+    loc = pvlib.location.Location(meta['latitude'], meta['longitude'], tz = meta['timezone'])
+    
+    power = normalized_daily2(times2()).resample('1h').interpolate()
+    poa = power*0+1000
+
+    rdSoiling = analysis.RdAnalysis(power,poa,ambient_temperature = power*0+25, pvlib_location=loc,
+                        temperature_coefficient=0, interp_freq='D' ) 
+    
+    rdSoiling.sensor_analysis(analyses=['srr_soiling'], srr_kwargs={'reps':reps})
+    
+    srr_results = rdSoiling.results['sensor']['srr_soiling']
+   
+    assert 0.9583 == pytest.approx(srr_results['p50_sratio'], abs=1e-4),\
+        'Soiling ratio different from expected value in RdAnalysis.srr_soiling'  
+    assert [0.9552, 0.9607] == pytest.approx(srr_results['sratio_confidence_interval'], abs=1e-4),\
+        'Soiling confidence interval different from expected value in RdAnalysis.srr_soiling' 
+    assert 0.97417 == pytest.approx(srr_results['calc_info']['renormalizing_factor'], abs=1e-4),\
+        'Renormalization factor different from expected value in RdAnalysis.srr_soiling' 
+"""
+
+    
+@pytest.fixture()
+def get_energy():
+    from degradation_test import DegradationTestCase  #re-use degradation test fixture
+    rd = -0.05
+    input_freq = 'H'
+    corr_energy = DegradationTestCase.get_corr_energy(rd, input_freq) # degradation_test.py
+    return corr_energy
+
+@pytest.fixture()
+def getRd(get_energy):
+    loc = pvlib.location.Location(meta['latitude'], meta['longitude'], tz = meta['timezone'])
+    power = get_energy.tz_localize(meta['timezone'])    
+    rd = analysis.RdAnalysis(power, power*1000, ambient_temperature = power*0+25, temperature_coefficient=meta['tempco'],
+                pvlib_location=loc, pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'], 
+                interp_freq='D', temperature_model = {'a': -3.47, 'b': -0.0594, 'deltaT': 3}) # temperature_model = "open_rack_glass_glass"
+    return rd
+    
+@pytest.fixture()
+def getRdCS(get_energy):
+    loc = pvlib.location.Location(meta['latitude'], meta['longitude'], tz = meta['timezone'])
+    power = get_energy.tz_localize(meta['timezone'])
+    # initialize the RdAnalysis class
+    temp = analysis.RdAnalysis(power, power*1000, pvlib_location=loc, 
+                        pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'], 
+                        temperature_coefficient=0 )
+    # get clearsky expected
+    temp.clearsky_preprocess()  # get expected clear-sky values in clearsky_poa.
+    cs = temp.clearsky_poa
+    
+    rdCS = analysis.RdAnalysis(power*cs, cs, temperature_coefficient=meta['tempco'],
+            pvlib_location=loc, pv_tilt=meta['tilt'], pv_azimuth=meta['azimuth'], 
+            temperature_model = {'a': -3.47, 'b': -0.0594, 'deltaT': 3}
+            )
+    return rdCS
```

Comment:
What is the intent for this fixture?  If I understand it correctly, it puts together an RdAnalysis object where the measured POA is simulated clear-sky POA.  Why is that useful?

Reply:
OK, hear me out.  when you import the get_energy degradation test fixture, it's just a single downhill slope over 3 years. The clearsky filtering won't work if there's no daily irradiance.    Once you have the clearsky_poa by running through clearsky_preprocess() once then you can multiply cs_poa by the degradation signal to get something close enough to a real daily power and irradiance change that can pass through the clearsky analysis without getting rejected.